make-rules/shared-macros.mk

```diff
@@ -1244,6 +1244,24 @@ CFLAGS +=	$(CC_BITS)
 # Add compiler specific 'default' features
 CFLAGS +=	$(CFLAGS.$(COMPILER))
 
+# Enable CTF if desired
+ifeq ($(strip $(USE_CTF)),yes)
+# -gdwarf-4 is the newest DWARF revision understood by illumos libctf
+#
+# -gstrict-dwarf prevents the compiler from including newer DWARF features
+#
+# -msave-args makes it possible for debugging tools (specifically pstack and
+# mdb) to retrieve function arguments, and is required for the python pstack
+# hooks.
+CFLAGS += -g -gdwarf-4 -gstrict-dwarf -msave-args
```

Comment:
Is `-msave-args` supported by `clang`?

Reply:
-msave-args is  an illumos special for GCC.   Not sure if it's available for clang (I found an email thread about it at https://lists.llvm.org/pipermail/llvm-dev/2017-April/112431.html  - unclear if that has gone anywhere.

what's the right way to condition makefiles on gcc vs clang?

Reply:
The compiler specific `CFLAGS` should go to `CFLAGS.$(COMPILER)`, so `CFLAGS.gcc` in this case.  Currently we do not have native support for other compilers (like `clang`) using this infrastructure, so `CFLAGS` and `CFLAGS.gcc` should always match.  Such facility was primarily used for `studio` (as alternative of `gcc`) but support for that is gone.

I assume that `USE_CTF` won't work with non-gcc compiler, so maybe you should also make sure that `USE_CTF` cannot be set for different `COMPILER` than `gcc`.

Reply:
> I assume that USE_CTF won't work with non-gcc compiler, so maybe you should also make sure that USE_CTF cannot be set for different COMPILER than gcc.

It's not that it's impossible to make it work - it's just that you likely need a different set of options for best results.
I've added a check to ensure that you can't enable CTF with a different compiler without thinking about it..